test/integration/puppetlabs/puppetserver/testutils.clj

```diff
@@ -101,7 +136,31 @@
   [catalogs resource-type resource-title]
   (count (filter #(catalog-contains? % resource-type resource-title) catalogs)))
 
-(schema/defn ^:always-validate with-puppet-conf-files
+(defmacro with-puppet-conf-files
+  "A macro that can be used to wrap a single test with the setup and teardown
+  of with-puppet-conf.
+
+  Accepts a map with the following characteristics:
+
+  * keys are simple filenames (no paths) that will be created in puppet's $confdir
+  * values are paths to a test-specific file that should be copied to puppet's confdir,
+    a la `$confdir/<key>`.
+
+  For each entry in the map, copies the files into $confdir, runs the test function,
+  and then deletes all of the files from $confdir.
```

Comment:
Maybe reword this a bit. I spent a few confusing seconds thinking it would run the body separately for each file in the map. Something like `Copies each of the files from the map into $confdir...`